pkg/sql/gcjob/gc_job.go

```diff
@@ -195,22 +203,84 @@ func unsplitRangesInSpan(ctx context.Context, kvDB *kv.DB, span roachpb.Span) er
 	return nil
 }
 
+// unsplitRangesInSpanForSecondaryTenant unsplits any manually splits ranges
+// within a span using an implementation that is appropriate for a secondary
+// tenant.
+//
+// When operating in a secondary tenant, unsplitting is not guaranteed becuase:
+//
+//   - The tenant may no longer be allowed to unsplit.
```

Comment:
Random drive-by nit: I can understand the motivation to have a "split" capability, but I don't follow why we'd want to gate "unsplit" behind a capability.  Is there some way that "unsplit" would do anything but leave the cluster in a better (i.e. more efficient) state?

Reply:
I am having trouble reconstructing the entire conversation. But, I think at least part of the argument was that `split` is more dangerous than `unsplit` so you might want to allow `unsplit` while restricting `split` in response to some run-away tenant.

But, eh, I do kinda wonder if we would be better served by one capability. Especially since requests still can only impact ranges whose start key is inside the tenant's keyspace.